src/amo/components/RatingsByStar/index.js

```diff
@@ -0,0 +1,149 @@
+import invariant from 'invariant';
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
+
+import { fetchGroupedRatings } from 'amo/actions/reviews';
+import { withFixedErrorHandler } from 'core/errorHandler';
+import translate from 'core/i18n/translate';
+import LoadingText from 'ui/components/LoadingText';
+import Icon from 'ui/components/Icon';
+import type { GroupedRatingsType } from 'amo/api/reviews';
+import type { AppState } from 'amo/store';
+import type { AddonType } from 'core/types/addons';
+import type { ErrorHandlerType } from 'core/errorHandler';
+import type { I18nType } from 'core/types/i18n';
+import type { DispatchFunc } from 'core/types/redux';
+
+import './styles.scss';
+
+type Props = {|
+  addon: AddonType | null,
+|};
+
+type InternalProps = {|
+  ...Props,
+  dispatch: DispatchFunc,
+  errorHandler: ErrorHandlerType,
+  groupedRatings?: GroupedRatingsType,
+  i18n: I18nType,
+|};
+
+export class RatingsByStarBase extends React.Component<InternalProps> {
+  componentWillMount() {
+    // TODO: this is intended to load on the server (before mount) but it
+    // does not.
+    // See: https://github.com/mozilla/addons-frontend/issues/5854
+    this.loadDataIfNeeded(this.props);
+  }
+
+  componentWillReceiveProps(nextProps: InternalProps) {
+    this.loadDataIfNeeded(nextProps);
+  }
+
+  loadDataIfNeeded(props: InternalProps) {
+    const { addon, dispatch, errorHandler, groupedRatings } = props;
+    if (!errorHandler.hasError() && addon && !groupedRatings) {
+      dispatch(
+        fetchGroupedRatings({
+          addonId: addon.id,
+          errorHandlerId: errorHandler.id,
+        }),
+      );
+    }
+  }
+
+  renderBarValue(starCount: number) {
+    const { addon } = this.props;
+    invariant(addon, 'addon is required');
+
+    let width = 0;
+    if (addon.ratings && addon.ratings.count > 0) {
+      width = Math.round((starCount / addon.ratings.count) * 100);
+    }
+
+    return (
+      <div
+        className={makeClassName(
+          'RatingsByStar-bar',
+          'RatingsByStar-barValue',
+          {
+            'RatingsByStar-barValueLessThan100': width < 100,
+          },
+        )}
+        style={{ width: `${width}%` }}
+      />
+    );
+  }
+
+  render() {
+    const { addon, errorHandler, i18n, groupedRatings } = this.props;
+    const loading = !addon || !groupedRatings;
+
+    return (
+      <div className="RatingsByStar">
+        {errorHandler.renderErrorIfPresent()}
+        <div className="RatingsByStar-graph">
+          {[5, 4, 3, 2, 1].map((star) => {
+            let starCount;
+            let starCountNode;
+
+            if (!errorHandler.hasError()) {
+              if (groupedRatings) {
+                starCount = groupedRatings[star];
+              }
+
+              starCountNode = loading ? (
+                <LoadingText minWidth={95} />
+              ) : (
+                i18n.formatNumber(starCount || 0)
```

Comment:
I dont know if this matters but in the comp (I am going by [this](https://github.com/mozilla/addons-frontend/issues/5437)), the numbers don't show comma. We might actually want this but wanted to verify this.

Also, the font was slightly different - I'm showing 15px here (in the "comp"). I know it's not a drastic difference from what we have (14px) but I don't know how particularly we should be ...? What do you think?

Reply:
> the numbers don't show comma

Ah, I didn't notice that. We need localized numbers for languages like Arabic and I think in other languages it will help users parse the result. German users will be able to read this more naturally, for example: `1.234.567`

> Also, the font was slightly different - I'm showing 15px here

I went with the [closest font size variable](https://github.com/mozilla/addons-frontend/blob/master/src/core/css/inc/vars.scss#L21) just for consistency with the rest of the site. I think we need to re-sync those variables with the spec because they are off in other places too.

Reply:
okay I figured, just double checking things : )